File: go/clock/clock.go
package clock

import "fmt"

// New returns a new Clock
func New(hour, minute int) Clock {
	var minutes = (hour * 60) + minute
	minutes %= 24 * 60
	if minutes < 0 {
		minutes += 24 * 60
	}

	return Clock{minutes: minutes}
}

// Clock represents a clock object that stores minutes
type Clock struct {
	minutes int
}

// Add adds minutes to a clock and returns a new clock
func (c Clock) Add(minutes int) Clock {
	return New(0, c.minutes + minutes)
}

// Subtract subtracts minutes from a clock object and returns a new clock
func (c Clock) Subtract(minutes int) Clock {
	return New(0, c.minutes - minutes)
}

// String returns a string representation of the clock
// with hours 0-23 minutes 0-59
func (c Clock) String() string {
	return fmt.Sprintf("%02d:%02d", c.minutes/60, c.minutes%60)
}

File: python/clock/clock.py
class Clock:
    def __init__(self, hour, minute):
        self.hours, self.minutes = divmod((hour * 60 + minute) % (24 * 60), 60)

    def __repr__(self):
        return "{:02d}:{:02d}".format(self.hours, self.minutes)

    def __eq__(self, other):
        return str(other) == str(self)

    def __add__(self, minutes):
        return Clock(self.hours, self.minutes + minutes)

    def __sub__(self, minutes):
        return Clock(self.hours, self.minutes - minutes)
